tests/test_arch_cuda_image.py

```
@@ -82,19 +82,37 @@ def test_validate_script_has_off_gpu_and_required_gpu_modes(self):
         )
+        self.assertTrue(scripts, "no .zsh scripts were discovered")
+        for script in scripts:
+            with self.subTest(script=script):
+                subprocess.run(
+                    ["zsh", "-n", script],
+                    cwd=REPO_ROOT,
+                    text=True,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    check=True,
+                )
```

Ruff (0.16.1)

[warning] Prefer `capture_output` over sending `stdout` and `stderr` to `PIPE`

Replace with `capture_output` keyword argument

(UP022)